app/models/quote.rb

```diff
@@ -0,0 +1,99 @@
+class Quote < ActiveRecord::Base
+
+  def self.get_rates(request)
+    parsed_request = JSON.parse(request)
+    @quote = self.new
+    @quote.request = request
+    @origin = Quote.get_origin
+    @packages = Quote.get_packages(parsed_request["number_of_items"])
+    address = parsed_request["address"]
+
+    if self.check_zip(address).nil?
+      return @quote
+    end
+
+    @destination = Quote.get_destination(address) #change to address when not testing
+    if @destination.nil?
+      return @quote
+    end
+
+    begin
+      carrier_responses = {
+         "FedEx" => self.fedex_rates,
+         "USPS" => self.usps_rates
+      }
+
+      @quote.response = carrier_responses.to_json
+      @quote.status = 200
+      @quote.save
+      @quote
+
+    rescue
+      @quote.response = { "Error": "Bad Request" }.to_json
+      @quote.status = 400
+      @quote.save
+      @quote
+    end
+  end
+
+  def self.get_origin
+    ActiveShipping::Location.new(country: "US", state: "WA", city: "Seattle", zip: "98121")
+  end
+
+  def self.get_destination(address)
+    begin
+      ActiveShipping::Location.new(country: address["country"], state: address["state"], city: address["city"], zip: address["zip"])
+    rescue
+      self.throw_422_error
+      return nil
+    end
+  end
+
+  def self.get_packages(number_of_items)
+    number_of_items = number_of_items.to_i
+    number_of_packages = (number_of_items / 4.0).ceil
+    packages = []
+    number_of_packages.times do
+      packages << ActiveShipping::Package.new(7, [15, 10, 4.5], units: :imperial)
```

Comment:
by not getting too caught up in details like package dimensions, you ended up with more time to implement tests and other cool stuff. nice project management!